File: guess-number/guess.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define LINE_MAX 80

void guess(int);

int main(void)
{
    int n = 100;
    guess(n);
    return 0;
}

void guess(int n)
{
    srand(time(NULL));
    int number = rand() % n;
    char guessStr[LINE_MAX];
    int guess;
    int attempt = 0;

    printf("Guess a number betweed 1 and %d\n", n);

    while (attempt < 10) {
        attempt++;
        printf("Please type in your guess: ");
        if ((fgets(guessStr, LINE_MAX, stdin) == NULL) || guessStr[0] == '\n')
            continue;
        guess = atoi(guessStr);

        if (guess < number)
            printf("Higher number please\n");
        else if (guess > number)
            printf("Lower number please\n");
        else {
            printf("Bingo! You guessed the number in %d attempts\n", attempt);
            return;
        }
    }

    printf("Oops! You lose after %d attempts\n", attempt);
}

File: guess-number/guess.go
package main

import(
    "bufio"
    "fmt"
    "math/rand"
    "os"
    "strconv"
    "time"
)

func main() {
    n := 100
    guess(n);
}

func guess(n int) {
    rand.Seed(time.Now().Local().UnixNano())
    number := rand.Intn(n) + 1
    attempt := 0

    fmt.Printf("Guess a number between 1 and %d\n", n)

    scanner := bufio.NewScanner(os.Stdin)
    for attempt < 10 {
        attempt += 1

        fmt.Print("Please type in your guess: ")
        scanner.Scan()
        guess, err := strconv.Atoi(scanner.Text())
        if err != nil {
            continue
        }

        if guess < number {
            fmt.Println("Higher number please")
        } else if guess > number {
            fmt.Println("Lower number please")
        } else {
            fmt.Printf("Bingo! You guessed the number in %d attempts\n", attempt)
            return
        }
    }

    fmt.Printf("Oops! You lose after %d attempts\n", attempt)
}
